modules/tinymce/src/core/main/ts/api/EditorOptions.ts

```diff
@@ -0,0 +1,277 @@
+   */
+  set: {
+    <K extends keyof EditorOptions>(name: K, value: EditorOptions[K]): boolean;
+    <T>(name: string, value: T): boolean;
+  };
+
+  /**
+   * Clear any set value for the specified option so that it'll revert back to using the default value.
+   *
+   * @method unset
+   * @param {String} name Name of a registered option.
+   * @return {Boolean} True if the option value was successfully reset, otherwise false.
+   */
+  unset: (name: string) => boolean;
+}
+
+const getBuiltInProcessor = <K extends BuiltInOptionType>(type: K): Processor<BuiltInOptionTypeMap[K], BuiltInOptionTypeMap[K]> => {
+  const validator = (() => {
+    switch (type) {
+      case 'array':
+        return Type.isArray;
+      case 'boolean':
+        return Type.isBoolean;
+      case 'function':
+        return Type.isFunction;
+      case 'number':
+        return Type.isNumber;
+      case 'object':
+        return Type.isObject;
+      case 'string':
+        return Type.isString;
+      case 'string[]':
+        return (val) => Type.isArrayOf(val, Type.isString);
+    }
+  })();
+
+  return (value) => ({ value, valid: validator(value), message: `The value must be of type: ${type}.` });
+};
+
+const isBuiltInSpec = <K extends BuiltInOptionType>(spec: unknown): spec is BuiltInOptionSpec<K> =>
+  Type.isString((spec as BuiltInOptionSpec<K>).processor);
+
+const getErrorMessage = (message: string, result: ProcessorError): string => {
+  const additionalText = Strings.isEmpty(result.message) ? '' : `. ${result.message}`;
+  return message + additionalText;
+};
+
+const isValidResult = <T>(result: ProcessorSuccess<T> | ProcessorError): result is ProcessorSuccess<T> =>
+  result.valid;
+
+const processValue = <T, U>(value: T, processor: SimpleProcessor<T> | Processor<T, U>): ProcessorSuccess<U> | ProcessorError => {
+  const result = processor(value);
+  if (Type.isBoolean(result)) {
+    // Note: Need to cast here as if a boolean is returned then we're guaranteed to be returning the same value
+    return result ? { value: value as unknown as U, valid: true } : { valid: false, message: '' };
+  } else {
+    return result;
+  }
+};
+
+const processDefaultValue = <T, U>(name: string, defaultValue: T, processor: Processor<T, U>): U | undefined => {
+  if (!Type.isUndefined(defaultValue)) {
+    const result = processValue(defaultValue, processor);
+    if (isValidResult(result)) {
+      return result.value;
+    } else {
+      // eslint-disable-next-line no-console
+      console.error(getErrorMessage(`Invalid default value passed for the "${name}" option`, result));
+    }
+  }
+
+  return undefined;
+};
+
+const create = (editor: Editor, initialOptions: Record<string, unknown>): Options => {
+  const registry: Record<string, OptionSpec<any, any>> = {};
+  const values: Record<string, any> = {};
+
+  editor.on('init', () => {
+    const unregisteredOptions = Arr.partition(Obj.keys(initialOptions), isRegistered).fail;
```

Comment:
Why use a partition if you only care about the fail case. Use `Arr.filter(Obj.keys(initialOptions), not(isRegistered))` instead. But I guess it would have been nice to have an Arr.reject or something that is the inverse of Array.filter.

Reply:
Yeah I did consider that, but I preferred this way of handling it, though I don't have any strong reasons. I'll swap to using `Arr.filter` with `Fun.not` then.

Reply:
Just feels wrong to use a branching function and not care about one of the sides making an array that not used at all. But maybe we should consider having an array filter that is the inverse out of scope here but might be useful.